### DISCUSSION TITLE
View newly summited row  in table without refreshing page.

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2728] [Help] [answered]
I am using a MYSQL connection that I am writing to with the form. At the bottom of the form, I have a data table that displays the rows submitted to the DB from that form. currently, I have to refresh the page to view the new row that was submitted. is there a way that I do not have to refresh the page to see the new row? 

I have this working with a relational DB setup. I select the data source as a relationship and it will populate the table without refreshing. however, I'm not sure how to do this without selecting a relationship as a data source.   

any help would be greatly appreciated.

**Comment by aptkingston** [comment]
Hey @KernOil. As you've seen, automatic refreshing of data is something that's built in to apps automatically. It uses the table IDs and data source IDs that your data providers are hooked up to, and whenever an action happens to one of those data sources (such as saving a row or deleting a row) it will automatically invalidate them and force a refresh. As for why it's not working with your current setup - I'm not sure, but I'm going to try and replicate it myself.

On another note, just yesterday I added a new feature for manually refreshing data sources on a button action (https://github.com/Budibase/budibase/pull/2730), so once that is released you would be able to manually trigger a refresh of your data provider after saving a row. This new feature is meant for things like REST data sources where automatic refreshing doesn't work due to table IDs not existing, but it's also a bandaid for issues like this. Automatic refreshing should work for any actual database sources though, so I'm treating this as a bug and I'll look into it.

**Reply by KernOil** [reply]
@aptkingston Thank you for the response. let me know if you need any more info from me when you are trying to replicate the problem.
I like the idea of a dedicated refresh button for the table. It would be a good stepping stone until all tables are refreshed properly.

**Comment by aptkingston** [comment]
I've tried to reproduce this, but refreshing automatically works fine. I'll show you my setup, and maybe you could provide screenshots of yours to see what the problem is.

I've have a data provider pulling from a table Tasks:
![image](https://user-images.githubusercontent.com/9075550/134648493-1d2e6100-93f0-489c-8556-5a5dc389833b.png)

I have a form on the same page to create a new Tasks row:
![image](https://user-images.githubusercontent.com/9075550/134648577-66cff2ee-1d01-432a-bf30-3d599563cef0.png)

And my button action saves a new Tasks row:
![image](https://user-images.githubusercontent.com/9075550/134648631-ba2c9a5a-ebd0-4aa3-aa00-ce77ef440b84.png)

This correctly automatically refreshes:
![image](https://user-images.githubusercontent.com/9075550/134648728-67ab38fc-2873-4fa2-a744-aff0b33e4e9e.png)

Does your configuration differ from this?

**Comment by KernOil** [comment, marked as answer]
Hi, @aptkingston sorry for the delay. my setup is the same as yours I am using an SQL DB that is the only thing I can think of that might be different.
![image](https://user-images.githubusercontent.com/80058238/136438046-b25ee475-1801-40d6-a2fc-71f0e2e9172c.png)
![image](https://user-images.githubusercontent.com/80058238/136438148-0169a148-11db-40db-b0de-fd5a46127755.png)
![image](https://user-images.githubusercontent.com/80058238/136438185-1b96f3c3-901f-4aa4-8a54-79f753bc7536.png)

https://user-images.githubusercontent.com/80058238/136438454-739c78fb-1b13-4f20-99e6-13503d2c6b18.mp4

**Reply by mjashanks** [reply]
Hey @KernOil - just following up on this. 2 things
1. We have now added a "Refresh Data Source" actions, so you can force this manually, so long as your Button is inside the dataprovider that you want to refresh
2. Is your Data Provider reading from a `Table`? i.e. **not** a custom query, as this will not work (you will have to use manual refresh in this case)

**Reply by KernOil** [reply]
This is perfect! Thank you. This will help out my application a lot, not having to refresh the page to see the new row!

**Reply by KernOil** [reply]
Got it working perfectly thank you both.
